static/compiler-picker.js

```diff
@@ -133,9 +138,9 @@ CompilerPicker.prototype.initialize = function (langId, compilerId) {
 CompilerPicker.prototype.getOptions = function (langId, compilerId) {
     var favorites = this.getFavorites();
     return _.chain(this.compilerService.getCompilersForLang(langId))
-        .filter(function (e) {
-            return !e.hidden || e.id === compilerId;
-        })
+        .filter(_.bind(function (e) {
+            return this.compilerIsVisible(e) && (!e.hidden || e.id === compilerId);
```

Comment:
(specifically:

return (this.compilerIsVisible && ! e.hidden) || e.id === compilerId;
)